image/src/utils/csv.js

```diff
@@ -0,0 +1,80 @@
+const fetch = require('node-fetch')
+const { FILENAME, DELIMITER, LIMIT_TYPE, LIMIT_SIZE, EGRESS_URL } = require('../config/config')
+const fs = require('fs')
+const { isValidURL } = require('./util')
+
+const processPayload = async json => {
+  const data = json.data !== undefined ? json.data : json
+  if (fs.existsSync(FILENAME)) {
+    const stats = fs.statSync(FILENAME)
+    const size = stats.size
+    const fileData = fs.readFileSync(FILENAME)
+    const lines = fileData.toString().split('\n').length
+    if ((LIMIT_TYPE === 'size' && size > LIMIT_SIZE) || (LIMIT_TYPE === 'rows' && lines > LIMIT_SIZE)) await upload()
+    await writeData(data, true)
+  } else {
+    await writeData(data, false)
+  }
+}
+
+const writeData = async (data, append) => {
+  let d = ','
+  switch (DELIMITER) {
+    case 'comma':
+      d = ','
+      break
+    case 'tab':
+      d = '\t'
+      break
+    case 'semicolon':
+      d = ';'
+      break
+  }
+  const logger = fs.createWriteStream(FILENAME, {
+    flags: 'a',
+  })
+  const writeLine = line => logger.write(`${line}\n`)
+  const keys = Object.keys(data[0])
+  console.log(keys)
+  let l = ''
+  if (!append) {
+    for (let i = 0; i < keys.length; i++) {
+      l += `${keys[i]}${d}`
+    }
+    writeLine(l)
+  }
+  l = ''
+  for (let i = 0; i < data.length; i++) {
+    for (let k = 0; k < keys.length; k++) l += `${data[i][keys[k]]}${d}`
+    writeLine(l)
+    l = ''
+  }
+  logger.end()
+}
+
+const upload = async () => {
+  let readStream = fs.createReadStream(FILENAME)
+  const stats = fs.statSync(FILENAME)
+  const size = stats.size
+  if (isValidURL(EGRESS_URL)) {
+    const res = await fetch(EGRESS_URL, {
+      method: 'POST',
+      headers: {
+        'Content-length': size,
+      },
+      body: readStream,
+    })
```

Comment:
According to the specs https://weevenetwork.notion.site/CSV-b6bc402bfe474d42a3b4db84b0fe33aa this module does not send file to the next module but it should `The output of the module should be a .csv file in the local file system.`. 
Please check how to mount a docker container with a host machine to save files in the local file system, otherwise you will be saving files in the container's file system that is not accessible from the outside (you may lose data).

Reply:
that was an old description, we will be storing to local within docker instance file. in future we can introduce redis or any other forms of caching.